src/main/scala/com/amazon/deequ/analyzers/MaxLength.scala

```diff
@@ -48,15 +51,35 @@ case class MaxLength(column: String, where: Option[String] = None, analyzerOptio
 
   override def filterCondition: Option[String] = where
 
-  private def criterion(nullBehavior: NullBehavior): Column = {
-    val isNullCheck = col(column).isNull
+  private[deequ] def criterion: Column = {
+    transformColForNullBehavior(col(column), getNullBehavior)
+  }
+
+  private[deequ] def rowLevelResults: Column = {
+    val filteredRowOutcome = getRowLevelFilterTreatment(analyzerOptions)
+    transformColForFilteredRow(criterion, filteredRowOutcome)
+  }
+
+  private def transformColForFilteredRow(col: Column, rowOutcome: FilteredRowOutcome): Column = {
+    val whereNotCondition = where.map { expression => not(expr(expression)) }
+    rowOutcome match {
+      case FilteredRowOutcome.TRUE =>
+        conditionSelectionGivenColumn(col, whereNotCondition, replaceWith = Double.MinValue)
+      case _ =>
+        conditionSelectionGivenColumn(col, whereNotCondition, replaceWith = null)
+    }
+  }
+
+  private def transformColForNullBehavior(col: Column, nullBehavior: NullBehavior): Column = {
```

Comment:
Can we use `column` here directly instead of passing it in as`col` ? In `rowLevelResults`, we use `analyzerOptions` directly instead of passing it in as a parameter.